Assets/__Scripts/MapEditor/AudioTimeSyncController.cs

```diff
@@ -192,6 +195,16 @@ private void OnDestroy()
         }
         if (OnPlayToggle != null) OnPlayToggle(IsPlaying);
     }
+    public void CancelPlaying()
+    {
+        IsPlaying = false;
+        CurrentSeconds = playStartTime;
+
+        songAudioSource.Stop();
+        SnapToGrid();
+
+        if (OnPlayToggle != null) OnPlayToggle(IsPlaying);
+    }
```

Comment:
I feel like this is re-inventing the wheel a little bit. Surely this method can't be replaced with a `TogglePlaying` call, then followed by setting `CurrentSeconds`? Maybe add an `IsPlaying` check just to be extra sure.

Reply:
Sorry it took so long to get back to ya, it's been a busy week.
I vaguely remember doing it this way in an effort to maintain the same order of operations. But present me isn't sure what past me was thinking.
Would it make more sense to reset CurrentSeconds _before_ calling TogglePlaying though? I've tested both ways, either works, but to prevent any future changes to SnapToGrid or anything listening to OnPlayToggle that might check CurrentSeconds from breaking.

Reply:
Well, what happens if I'm in the middle of a beat but set my precision to 1/1 and then press, hold and release?
Surely depending on the order here I'll either return to the middle of the beat or return to a snapped beat?

Reply:
Ahhh, SnapToGrid, gotcha. You are correct. I would say returning to exactly where I was would be the preferred option. For the same reason that if I change my precision, I don't immediately get snapped to the grid - it's unexpected behavior if anything other than myself changes the position.